include/nbl/system/ISystem.h

```diff
@@ -23,7 +23,7 @@ class ISystem final : public core::IReferenceCounted
 
     public:  
         virtual core::smart_refctd_ptr<IFile> createFile(ISystem* sys, const std::filesystem::path& filename, IFile::E_CREATE_FLAGS flags) = 0;
-        virtual core::smart_refctd_ptr<CMemoryFile> createBufferedFile(ISystem* sys, const std::filesystem::path& filename, IFile::E_CREATE_FLAGS flags) = 0;
+        virtual core::smart_refctd_ptr<CFileView> createBufferedFile(ISystem* sys, const std::filesystem::path& filename, IFile::E_CREATE_FLAGS flags) = 0;
```

Comment:
dont even need a factory for this, CFileView can create itself no problem

Reply:
lets have this function in ISystem API and public ctor in CFileView as well (CFileView.h should be in `include/` ofc)

Reply:
also you dont need it to be `virtual`